Update frame metrics / viewport when orientation changes

rainemak opened the pull request:
Sending zoom contraints update from TabChildHelper is delayed
until new target viewport size is calculated. The new target
viewport size is used for calculating minimum viewport size.

AsyncPanZoomController also resets displayPort
when composition bounds have changed.

Viewport related logs added behind "EmbedLiteViewPort" logging component.

tmeshkova commented:
Could you split this into functional part and logging part, also for logging part move that huge log function LOGC("EmbedLiteViewPort", ..... into one place with metrics argument.

Commit:
Update frame metrics / viewport when orientation changes

Sending zoom contraints update from TabChildHelper is delayed
until new target viewport size is calculated. The new target
viewport size is used for calculating minimum viewport size.

In addition, AsyncPanZoomController needs to reset also displayPort
when composition bounds have changed.

Commit:
Add more framemetrics logging for AsyncPanZoomController

Commit:
Add new LOG_FM macro for debugging frame metrics

Add "EmbedLiteViewPort:4" to NSPR_LOG_MODULES environment
variable to output frame metrics from embedlite.

Commit:
Add window size related logs behind LOGT

Add "EmbedLiteTrace:4" to NSPR_LOG_MODULES environment
variable to see those.

rainemak commented:
Fixed small typo that slipped into the LOG_FM macro
